### ISSUE TITLE
Observer builds null when something inside its builder throws

### ISSUE THREAD

**Issue by izik1** [mobxjs/mobx.dart#202]
minimum example:
```dart
class FooWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) =>
      Observer(builder: (context) => throw Exception());
}
```

I believe this to be because `ReactiveContext.trackDerivation` (called by `ReactionImpl.track`) stores the `Exception`. This eventually leads to `ReactionImpl._reportException` which may or may not throw. If it does throw, the same problem as in #197 occurs. If it _doesn't_, `Observer` builds with `built = null`, it seems that in debug mode at least, the latter happens.

**Comment by pavanpodila** [comment]
Looks like there is no clear answer right now. MobX has a setting: `ReactiveConfig#disableErrorBoundaries`, which will surface the exception happening inside reactions. The reason why MobX catches it is to keep the reactive-system consistent and unwrap all tracked reactions. 

What I would suggest is to try with a custom ReactiveConfig on the `mainContext`, where `disableErrorBoundaries = true`. Use this during development and then put it back to the original value.

```dart
mainContext.config = ReactiveConfig(disableErrorBoundaries: true);

// Later
mainContext.config = ReactiveConfig.main;
```

**Comment by pavanpodila** [comment]
@izik1 Although we thought the solution developed as part of #197 was good, it was causing the issue you raised. For now, we have reverted the solution to throw the exception raised during the tracking of the reaction. It is published as [0.2.3](https://pub.dev/packages/flutter_mobx).
